src/components/Desktop/Window/Window.tsx

```diff
@@ -152,15 +87,94 @@ export const Window = ({ appID }: WindowProps) => {
       minHeight="300"
       onDragStart={focusCurrentApp}
     >
-      <section className={css.container} tabIndex={-1} ref={containerRef} onClick={focusCurrentApp}>
-        <div>
-          <header className={clsx({ 'app-window-drag-handle': true, [css.titleBar]: true })}>
-            <TrafficLights appID={appID} onMaximizeClick={maximizeApp} />
-          </header>
-          <div className={css.divider} />
-        </div>
-        <PlaceholderApp appID={appID} />
+      <section
+        className={clsx(css.container, theme === 'dark' && css.dark)}
+        tabIndex={-1}
+        ref={containerRef}
+        onClick={focusCurrentApp}
+      >
+        <AppNexus appID={appID} />
       </section>
     </Rnd>
   );
 };
+
+/**
+ * Extract the x and y from the transform style of the base element using Regex
+ * Why using this hacking method:
+ * react-rnd uses transform and translate to shift window around instead of top
+ * and left and it does not provide the access to x and y values from ref
+ * @param transformStyle The transform style string. e.g. translate(1123.75px, 7px)
+ * @returns The window position. e.g. { x: 1123.75, y: 7 }
+ */
+function extractPositionFromTransformStyle(transformStyle: string): WindowPosition {
+  const matched = transformStyle.matchAll(/[0-9.]+/g);
+  try {
+    return { x: Number(matched.next().value[0]), y: Number(matched.next().value[0]) };
+  } catch {
+    return { x: 0, y: 0 };
+  }
+}
+
+const useMaximizeWindow = (
+  windowRef: RefObject<WindowRnd>,
+  transitionClearanceRef: RefObject<number>,
```

Comment:
Actually `transitionClearanceRef` is just a simple ref to track the `setTimeout`, which doesn't need to be passed in. I think you can just create this as a `useRef` in `useMaximizeWindow` itself

Reply:
Great. Will do that

Reply:
Hehe, actually only the windowRef was needed as external ref, rest all could be moved into the hook itself